google-maps-workitem/src/main/java/org/jbpm/workitem/google/maps/GoogleMapsGeocodingWorkitem.java

```diff
@@ -0,0 +1,164 @@
+        String searchLocation = (String) workItem.getParameter("SearchLocation");
+        String bounds = (String) workItem.getParameter("Bounds");
+        String locationType = (String) workItem.getParameter("LocationType");
+
+        try {
+
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+            if (geoApiContext == null) {
+                geoApiContext = auth.authorize(apiKey);
+            }
+
+            GeocodingApiRequest geocodingApiRequest = GeocodingApi.newRequest(geoApiContext);
+            Map<String, Object> results = new HashMap<>();
+
+            switch (searchType) {
+                case "byaddress":
+                    geocodingApiRequest = geocodingApiRequest.address(searchLocation);
+                    break;
+                case "byplaceid":
+                    geocodingApiRequest = geocodingApiRequest.place(searchLocation);
+                    break;
+                case "bylatlong":
+                    try {
+                        String[] searchLocationParts = searchLocation.split(",");
+                        geocodingApiRequest = geocodingApiRequest.latlng(new LatLng(Double.parseDouble(searchLocationParts[0]),
+                                                                                    Double.parseDouble(searchLocationParts[1])));
+                    } catch (Exception e) {
+                        throw new IllegalArgumentException("Unable to perform search by latlong - invalid location: " + searchLocation);
+                    }
+                    break;
+                default:
+                    // default to address search
+                    geocodingApiRequest = geocodingApiRequest.address(searchLocation);
+            }
+
+            if (bounds != null && bounds.length() > 0) {
+                String[] boundsParts = bounds.split(",");
+                if (boundsParts.length == 4) {
+                    try {
+                        geocodingApiRequest = geocodingApiRequest.bounds(new LatLng(Double.parseDouble(boundsParts[0]),
+                                                                                    Double.parseDouble(boundsParts[1])),
+                                                                         new LatLng(Double.parseDouble(boundsParts[2]),
+                                                                                    Double.parseDouble(boundsParts[3])));
+                    } catch (NumberFormatException e) {
+                        logger.error("Invalid bounds format: " + bounds + ". Valid example could be: 34.172684,-118.604794,34.236144,-118.500938");
+                    }
+                } else {
+                    logger.error("Invalid bounds: " + bounds + ". Valid example could be: 34.172684,-118.604794,34.236144,-118.500938");
+                }
+            }
+
+            if (locationType != null) {
```

Comment:
All parameters look quite complex so might be worth to document it, wdyt?

Reply:
good idea. i added description of parameters that can be multiple values of all handlers in the readme file